client/src/js/collections/DogsCollection.js

```diff
@@ -1,14 +1,55 @@
+var $ = require('jquery');
 var Backbone = require('backbone');
 
+var statusMap = {
+  0: {
+    color: 'green',
+    label: 'in office'
+  },
+
+  1: {
+    color: 'yellow',
+    label: 'on a walk'
+  },
+
+  2: {
+    color: 'red',
+    label: 'missing'
+  }
+};
+
 var Dog = Backbone.Model.extend({
   defaults: {
     display_id: 0,
     name: '',
     short_description: '',
     checked_in: false,
-    avatar: ''
+    avatar: '',
+    current_status: 0
+  },
+  idAttribute: 'display_id',
+
+  initialize: function () {
+    this.setVerboseStatus();
+
+    this.on('change:current_status', this.setVerboseStatus);
```

Comment:
I'm not sure why we're calling `setVerboseStatus` in these places in the `initialize` function.

`setVerboseStatus` just returns a tuple of `label, color`, it doesn't actually assign anything. I think these calls are pointless?